programs/drift/src/controller/liquidation.rs

```diff
@@ -1737,6 +1756,10 @@ pub fn liquidate_perp_pnl_for_deposit(
         None,
     )?;
 
+    let (safest_tier_spot_liability, safest_tier_perp_liability) =
+        calculate_user_safest_position_tiers(user, perp_market_map, spot_market_map)?;
+    let is_contract_tier_violation =
```

Comment:
feel like it might just be cleaner to check the tier violation here and return early all at once here, instead of spreading the logic, but will defer to you

Reply:
that'd be a lot of copy paste code tho